tests/test_package.py

```diff
@@ -59,45 +73,188 @@ def test_share_git_package():
+        "torchaudio": f"torchaudio {cuda_116_url} {extra_index_url}",
+        "matchatorch": f"matchatorch {cuda_116_url} {extra_index_url}",
+        "torch==1.13.0": f"torch==1.13.0 {cuda_116_url} {extra_index_url}",
+        f"torch {cuda_116_url}": f"torch {cuda_116_url} {extra_index_url}",
+        f"torch>=1.13.0 {cuda_116_url}": f"torch>=1.13.0 {cuda_116_url} {extra_index_url}",
+    }
+
+    # Write these packages to a temp reqs file, then update each one with their full install URLs
+    package_names = list(packages_install_cmds)
+    with open(test_reqs_file, "w") as f:
+        f.writelines([line + "\n" for line in package_names])
+
+    dummy_pkg = rh.Package.from_string(specifier="pip:dummy_package")
 
-    extra_index_url = "--extra-index-url https://pypi.python.org/simple/"
+    # This mocks the logic from: `format_torch_cmd_in_reqs_file`
+    with open(test_reqs_file) as f:
+        reqs = f.readlines()
 
-    # Receive a torch install command and associated cuda version, confirm that `install_cmd_for_torch` appends
-    # the correct --index-url for the matching cuda version
+        for req in reqs:
+            req = req.strip()
+            install_cmd = dummy_pkg.install_cmd_for_torch(req, cuda_version="11.6")
+            assert install_cmd == packages_install_cmds[req]
+
+    test_reqs_file.unlink()
+
+    assert True
+
+
+@pytest.mark.localtest
+def test_torch_install_command_generator():
+    """Checks that the command itself is correct (without actually running it on the cluster)"""
+    cuda_version = "11.6"
     install_commands = {
-        (f"torch==1.13.1 {cuda_v116_url}", "11.6"): f"torch==1.13.1 {cuda_v116_url} {extra_index_url}",
-        (f"torch {extra_index_url}", "11.7"): f"torch {extra_index_url} {cuda_v117_url}",
-        ("matchatorch", "11.6"): f"matchatorch {cuda_v116_url} {extra_index_url}",
-        ("torch>=1.13.0, <2.0.0", "11.8"): f"torch>=1.13.0, <2.0.0 {cuda_v118_url} {extra_index_url}",
-        (f"torch {cuda_v116_url} torchaudio {extra_index_url}", "11.6"): f"torch {cuda_v116_url} {extra_index_url} "
-                                                                         f"torchaudio {extra_index_url} "
-                                                                         f"{cuda_v116_url}",
-        ("torch torchpudding", "11.7"): f"torch {cuda_v117_url} {extra_index_url} torchpudding {cuda_v117_url} "
-                                        f"{extra_index_url}",
-        ("torch>=1.13.0", "11.6"): f"torch>=1.13.0 {cuda_v116_url} {extra_index_url}",
-        ("torch>1.13.0", "11.6"): f"torch>1.13.0 {cuda_v116_url} {extra_index_url}",
-        ("torch==1.13.0", "11.6"): f"torch==1.13.0 {cuda_v116_url} {extra_index_url}",
-        ("torch~=1.13.0", "11.7"): f"torch~=1.13.0 {cuda_v117_url} {extra_index_url}",
-        (f"torch==99.99.999 {cuda_v117_url}", "11.7"): f"torch==99.99.999 {cuda_v117_url} {extra_index_url}",
+        (
```

Comment:
This is just a silly formatting nit, but this doesn't need to be a dict, it can just be a list of lists and then it'd probably be a lot more readable